### PR TITLE
B-19178 MTO shipment risk of excess banner persistence solution

### PR DESCRIPTION
**Pull request by landan-parker-0**
## Summary

Is there anything you would like reviewers to give additional scrutiny?
Continuation of:
https://github.com/transcom/mymove/pull/12499

### CODE REVIEW

File: src/pages/Office/CustomerOnboarding/CreateCustomerForm.test.jsx [hunk 1]
```diff
@@ -171,6 +171,58 @@ describe('CreateCustomerForm', () => {
...
+    const waiter = waitFor(() => {
+      expect(createCustomerWithOktaOption).toHaveBeenCalled();
+    });
+
+    await user.click(saveBtn);
+    await waiter;
+    expect(mockNavigate).toHaveBeenCalled();
+  }, 10000);
+
+  it('submits the form and tests for unsupported state validation', async () => {
```
Comment (pambecker) [inline src/pages/Office/CustomerOnboarding/CreateCustomerForm.test.jsx:204]: I noticed this file had more changes than your INT PR.  seems this test is duplicated in this file.  maybe a bad merge fix

File: src/pages/Office/CustomerOnboarding/CreateCustomerForm.test.jsx [hunk 2]
```diff
@@ -171,30 +171,33 @@ describe('CreateCustomerForm', () => {
...
+    await user.type(getByTestId('backup-add-city'), fakePayload.backup_mailing_address.city);
+    await user.selectOptions(getByTestId('backup-add-state'), [fakePayload.backup_mailing_address.state]);
+    await user.type(getByTestId('backup-add-zip'), fakePayload.backup_mailing_address.postalCode);
 
-    await userEvent.type(getByTestId('create-okta-account-yes'), fakePayload.create_okta_account);
+    await user.type(getByLabelText('Name'), fakePayload.backup_contact.name);
+    await user.type(getByRole('textbox', { name: 'Email' }), fakePayload.backup_contact.email);
+    await user.type(getByRole('textbox', { name: 'Phone' }), fakePayload.backup_contact.telephone);
 
-    await userEvent.type(getByTestId('cac-user-no'), fakePayload.cac_user);
```
Comment (pambecker) [inline src/pages/Office/CustomerOnboarding/CreateCustomerForm.test.jsx:190]: did you mean to delete this line?

File: src/pages/Office/CustomerOnboarding/CreateCustomerForm.test.jsx [hunk 3]
```diff
@@ -171,30 +171,33 @@ describe('CreateCustomerForm', () => {
...
 
-    await userEvent.type(getByLabelText('Name'), fakePayload.backup_contact.name);
-    await userEvent.type(getByRole('textbox', { name: 'Email' }), fakePayload.backup_contact.email);
-    await userEvent.type(getByRole('textbox', { name: 'Phone' }), fakePayload.backup_contact.telephone);
+    await user.type(getByTestId('backup-add-street1'), fakePayload.backup_mailing_address.streetAddress1);
+    await user.type(getByTestId('backup-add-city'), fakePayload.backup_mailing_address.city);
+    await user.selectOptions(getByTestId('backup-add-state'), [fakePayload.backup_mailing_address.state]);
+    await user.type(getByTestId('backup-add-zip'), fakePayload.backup_mailing_address.postalCode);
 
-    await userEvent.type(getByTestId('create-okta-account-yes'), fakePayload.create_okta_account);
```
Comment (pambecker) [inline src/pages/Office/CustomerOnboarding/CreateCustomerForm.test.jsx:188]: Maria updated this line (188) and line 190 in a recent merge to main, be sure you aren't breaking any logic.  I think we need to keep these lines.

**Review by pambecker — COMMENTED** [review]
Please delete the extra branches not being used for this backlog.

**Review by pambecker — APPROVED** [review]
this matches the INT PR.  Please delete the extra 19178 int branch as mentioned before.

**Review by loganwc — APPROVED** [review]

File: pkg/services/move/move_weights.go [hunk 4]
```diff
@@ -116,35 +110,43 @@ func (w moveWeights) CheckExcessWeight(appCtx appcontext.AppContext, moveID uuid
...
+	var shouldSaveMoveRecord = true
+	if shouldExcessWeightBecomeQualified && move.ExcessWeightAcknowledgedAt == nil {
+		move.ExcessWeightQualifiedAt = &now
+	} else if !shouldExcessWeightBecomeQualified {
+		move.ExcessWeightQualifiedAt = nil
+		move.ExcessWeightAcknowledgedAt = nil
+	} else {
+		shouldSaveMoveRecord = false
+	}
+	if shouldSaveMoveRecord {
```
Comment (paulstonebraker) [inline pkg/services/move/move_weights.go:149]: Were there any logic changes or was this refactored for some reason? I don't see any changes

File: src/pages/Office/MoveTaskOrder/MoveTaskOrder.jsx [hunk 5]
```diff
@@ -388,6 +391,34 @@ export const MoveTaskOrder = (props) => {
       milmoveLogger.error(errorMsg);
     },
   });
+
+  useEffect(() => {
+    setIsAtExcessWeightRisk(hasRiskOfExcess(estimatedWeightTotal, order?.entitlement?.authorizedWeight));
+  }, [estimatedWeightTotal, order?.entitlement?.authorizedWeight]);
```
Comment (paulstonebraker) [inline src/pages/Office/MoveTaskOrder/MoveTaskOrder.jsx:397]: I thought this was accurate (the excess weight flag directs your to look review the max billable weight), but the way it was originally is accurate. We should be checking against entitlement.totalWeight.

`A move is considered having excess weight if the total move weight exceeds the customer's weight allowance. A move is considered having risk of excess weight if the estimated weight total is within 90% of the customer’s weight allowance.`

File: src/pages/Office/MoveDetails/MoveDetails.jsx [hunk 6]
```diff
@@ -118,6 +123,20 @@ const MoveDetails = ({
       setAlertType('error');
     },
   });
+  useEffect(() => {
+    setIsAtExcessWeightRisk(hasRiskOfExcess(estimatedWeightTotal, order?.entitlement?.authorizedWeight));
+  }, [estimatedWeightTotal, order?.entitlement?.authorizedWeight]);
```
Comment (paulstonebraker) [inline src/pages/Office/MoveDetails/MoveDetails.jsx:128]: Same comment as I left in MoveTaskOrder.jsx. I think we need to clear up with PO how we want to handle this

File: src/pages/Office/MoveTaskOrder/MoveTaskOrder.jsx [hunk 7]
```diff
@@ -136,8 +138,9 @@ export const MoveTaskOrder = (props) => {
 
   const { orders = {}, move, mtoShipments, mtoServiceItems, isLoading, isError } = useMoveTaskOrderQueries(moveCode);
   const order = Object.values(orders)?.[0];
-  const nonPPMShipments = mtoShipments?.filter((shipment) => shipment.shipmentType !== 'PPM');
```
Comment (paulstonebraker) [inline src/pages/Office/MoveTaskOrder/MoveTaskOrder.jsx:139]: not going to die on this hill but I think this is easier to read

**Review by paulstonebraker — CHANGES_REQUESTED** [review]

**Pull request closed without merging**